spec/classes/datadog_agent_integrations_haproxy_spec.rb

```diff
@@ -7,6 +7,7 @@
       let(:facts) do
         {
           ipaddress: '1.2.3.4',
```

Comment:
Eventually this should be removed, correct?
And the same for other spec files.

Reply:
I planned to remove the legacy Facts from the Rspec-Tests, as soon as the Tests succeed.  But currently they don't.

Honestly, I don't know, whats wrong.  In my own Projects I am used to use the Facts from the Gem `facterdb` like so: 
```
  on_supported_os.each do |os, os_facts|
    context "on #{os}" do
      let(:facts) { os_facts }

      it { is_expected.to compile.with_all_deps }
    end
  end
```

`os_facts` comes out of `facterdb`.  
`on_supported_os.each` iterates over all OS-Releases stated in `metadata.json`.   So you don't have to specify the Facts yourself for each OS's Context.

But here, the Tests are designed differently, which leaves me a bit lost…  
Perhaps you find the Problem.  I am a bit short of Time right now.  

I invested an Amount of Time in this Module.  More PRs are already prepared:  
* migrate to PDK 3 with most recent Version of PDK-Templates 
* fix all the Linters' Complaints

Reply:
Wow, that's cool.  I cherry-picked, rebased against `main` and force-pushed.  Checks seem okay now.

Reply:
you can maybe remove in a next PR, seems it raises new errors :/